### PR TITLE
Fix external-oauth example jupyterhub_config.py

### PR DESCRIPTION
**Pull request by yuvipanda**
- Roles need to be explicitly granted, otherwise you get a
  403. This example predates roles.
- Explicitly set bind_url - without this, JupyterHub itself doesn't seem to bind anywhere, and so you just get a 404 when you visit whatever port configurable-http-proxy lands on. This is probably a separate bug to be investigated, but in the meantime copying this from testing/jupyterhub_config.py makes this example actually work
- Set DummyAuthenticator as the default, so users can get started with this example

### CODE REVIEW

File: examples/external-oauth/jupyterhub_config.py [hunk 1]
```diff
@@ -1,5 +1,16 @@
 import os
 
+# Allow anyone to authenticate
```
Comment (minrk) [inline examples/external-oauth/jupyterhub_config.py:3]: In some other examples that have this, I've [put it at the bottom](https://github.com/jupyterhub/jupyterhub/blob/f4163069139251c2be2d7752607518e97bd31079/examples/server-api/jupyterhub_config.py#L54-L57) to separate it as clearly the "this isn't relevant to the topic of the example, but is needed to make the example runnable". I think that makes it easier to "stop reading here..."

**Review by minrk — APPROVED** [review]
👍 after moving the test boilerplate to the bottom

> without this, JupyterHub itself doesn't seem to bind anywhere

That would definitely be a separate bug, but I'm not sure there is one here. This config is setting the bind URL for the proxy itself. The Hub process or what is proxied has not been affected. The default is `http://:8000`, which binds to all available interfaces. My _guess_ is that perhaps you had something running on that port already on 127.0.0.1, and binding to `''` weirdly _succeeds_ if there are any other services bound to more specific interfaces, and only gets requests not matched by another, more specific listening socket.

So you can do:

```python
listen('127.0.0.1', port)
listen('1.2.3.4', port)
listen('', port)
```

and all will succeed, but the `''` will not get any requests unless you make requests to an ip not already bound by another one.

File: examples/external-oauth/jupyterhub_config.py (hunk 1 above)
Comment (yuvipanda) [reply examples/external-oauth/jupyterhub_config.py:3]: @minrk I've moved it to the bottom!

**Comment by yuvipanda** [comment]
@minrk I've also opened https://github.com/jupyterhub/jupyterhub/issues/4557 with repro steps for `bind_url` issues.

**Pull request merged**